### PR TITLE
Gap/Stagger Aug Removal: Test if σ=0 Beats Baseline Now That GSB Exists

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The gap/stagger augmentation (`--aug_gap_stagger_sigma 0.02`, PR #2115) was merged because it improved p_oodc by -4.9%. However, the original combined-baseline analysis showed it **hurts p_tan by +1.6%** relative to the no-aug baseline. This was accepted as a trade-off.

Since then, **Gap/Stagger Spatial Bias (GSB, PR #2130)** was merged, which makes slice routing tandem-geometry-aware by feeding gap+stagger directly into the spatial_bias MLP. GSB provides the model with explicit tandem geometry information at the routing level.

**The hypothesis:** GSB may have made the gap/stagger augmentation redundant. GSB already tells the model about tandem geometry — the augmentation just adds noise to the gap/stagger scalars, which could now be confusing rather than helpful. Removing the aug (σ=0) should:
1. Recover the +1.6% p_tan penalty
2. Possibly maintain p_oodc if GSB provides sufficient geometry awareness

This is complementary to askeladd #2140 (testing σ=0.01). Together we get a 3-point sweep: σ={0, 0.01, 0.02(baseline)}.

**Expected impact:** p_tan -0.5% to -2% (recovering the known aug penalty). Risk: p_oodc may regress if the aug was providing genuine robustness beyond what GSB offers.

### NO CODE CHANGES NEEDED — just remove the `--aug_gap_stagger_sigma` flag.

Run 2 seeds with the baseline command but WITHOUT `--aug_gap_stagger_sigma`:

```bash
# Seed 42
cd cfd_tandemfoil && python train.py --agent tanjiro \
  --wandb_name "tanjiro/gs-remove-s42" --wandb_group phase6/gs-aug-removal \
  --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias

# Seed 73
cd cfd_tandemfoil && python train.py --agent tanjiro \
  --wandb_name "tanjiro/gs-remove-s73" --wandb_group phase6/gs-aug-removal \
  --seed 73 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

**Note:** The ONLY difference from the baseline command is the ABSENCE of `--aug_gap_stagger_sigma 0.02`. Everything else is identical.

### What to report

| Config | Seed | p_in | p_oodc | p_tan | p_re | W&B |
|--------|------|------|--------|-------|------|-----|
| σ=0 (no aug) | 42 | | | | | |
| σ=0 (no aug) | 73 | | | | | |
| **σ=0 avg** | — | | | | | |
| **Baseline (σ=0.02)** | — | 13.05 | 7.70 | 28.60 | 6.55 | d7l91p0x, j9btfx09 |

Key question: does removing the aug improve p_tan without catastrophic p_oodc regression?

## Baseline

Current baseline (PR #2130, GSB + PCGrad):

| Metric | 2-seed target |
|--------|--------------|
| p_in   | < 13.05      |
| p_oodc | < 7.70       |
| p_tan  | **< 28.60**  |
| p_re   | < 6.55       |

W&B: d7l91p0x (s42), j9btfx09 (s73)

```bash
cd cfd_tandemfoil && python train.py --agent <name> --wandb_name "<name>/baseline-gsb-pcgrad" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

### Implementation

No code changes required. Ran baseline command with `--aug_gap_stagger_sigma` removed.

### Results Table

| Config | Seed | Best Epoch | val/loss | p_in | p_tan | p_oodc | p_re | W&B |
|--------|------|-----------|----------|------|-------|--------|------|-----|
| σ=0 (no aug) | 42 | 150 | 0.3823 | **13.0** | 29.5 | 7.7 | **6.4** | `3h9fj0ym` |
| σ=0 (no aug) | 73 | 151 | 0.3869 | 14.0 | 30.0 | 7.9 | **6.4** | `j40ez5v3` |
| **σ=0 avg** | — | — | — | 13.5 | **29.75** | **7.80** | 6.4 | — |
| **Baseline σ=0.02** | — | — | — | **13.05** | **28.60** | **7.70** | 6.55 | d7l91p0x, j9btfx09 |

### vs Baseline

| Metric | σ=0 avg | Baseline (σ=0.02) | Δ |
|--------|---------|-------------------|---|
| p_in | 13.5 | **13.05** | +3.4% ❌ |
| p_tan | 29.75 | **28.60** | **+4.0% ❌** |
| p_oodc | 7.80 | **7.70** | +1.3% ❌ |
| p_re | **6.4** | 6.55 | -2.3% ✓ |

### Command Used

```bash
cd cfd_tandemfoil && CUDA_VISIBLE_DEVICES=<gpu> python train.py --agent tanjiro \
  --wandb_name "tanjiro/gs-remove-s<seed>" --wandb_group phase6/gs-aug-removal \
  --seed <seed> \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_dsdf2_sigma 0.05 \
  --pcgrad_3way --pcgrad_extreme_pct 0.15 --gap_stagger_spatial_bias
```

Peak memory: ~38 GB.

### Analysis

**The hypothesis is false: removing gap/stagger augmentation hurts all primary metrics.** p_tan goes from 28.60 → 29.75 (+4.0%), p_oodc from 7.70 → 7.80 (+1.3%), p_in from 13.05 → 13.50 (+3.4%).

**Why GSB and aug are complementary, not redundant:**

1. **GSB (gap_stagger_spatial_bias) provides geometry-aware routing** — the model knows where to route computation based on gap/stagger values. But it only sees the training distribution of gap/stagger values.

2. **Aug provides regularization against gap/stagger distribution shift** — by perturbing gap/stagger during training, the model learns to be robust to unseen gap/stagger combinations. This is distinct from GSB's routing role.

3. The primary eval `val_tandem_transfer` (p_tan) tests NACA6416 as the front foil with the training distribution of gap/stagger. Without aug, the model overfits to exact training gap/stagger values and fails to extrapolate. With aug, it learns a smoother mapping.

4. p_re improved slightly without aug (6.4 vs 6.55), which might indicate the aug very slightly hurts the in-distribution Reynolds number cases. But this is much smaller than the p_tan regression.

**Conclusion:** The gap/stagger augmentation is still valuable even with GSB. They address different failure modes: GSB improves representation, aug improves robustness. Keep σ=0.02 in the baseline.

### Suggested Follow-ups

1. **Test σ=0.01 (askeladd #2140 is exploring this):** The aug at σ=0.02 might be a sweet spot, but a smaller σ could preserve the robustness benefit while reducing disruption to the learned representation. Combined with GSB, the optimal sigma might shift downward.
2. **Larger sigma (σ=0.05 or σ=0.10):** If the current σ=0.02 provides regularization benefit, a larger sigma might push robustness further — or degrade representation quality like DSDF1 aug.
3. **Independent analysis:** Compare {no-aug, GSB=False} vs {aug=0.02, GSB=False} to isolate aug's contribution without GSB, confirming it's the aug doing the work.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2148 because ADVISOR: Closing — clear dead end. Removing gap/stagger augmentation (σ=0) degrades all primary metrics vs baseline: p_in +3.4%, p_tan +4.0%, p_oodc +1.3%. Only p_re improved slightly (-2.3%), which is not enough.

Your analysis is spot-on: GSB and aug are complementary, not redundant. GSB handles geometry-aware routing; aug provides regularization against distribution shift. Both are needed. This also confirms that σ=0.02 is well-calibrated — we already tested σ=0.01 (#2140, worse) and now σ=0 (worse). The σ=0.02 sweet spot is robust.

Combined with the σ=0.01 result from #2140, we can now consider the entire gap/stagger sigma parameter space explored: {0, 0.01, 0.02(baseline), 0.03(frieren #2153 in progress)}. σ=0.02 is optimal or near-optimal.

Good work on the analysis — the insight about GSB handling representation vs aug handling robustness is valuable for future experiment design. Stand by for your next assignment..
